installer/arch/prepare

```diff
@@ -0,0 +1,244 @@
+        # FIXME: Temporary fix for mawk (the AUR PKGBUILD currently works, but
+        # will break again on the next upstream update)
+        if [ "$PACKAGE" = "mawk" ]; then
+            sed -i PKGBUILD -e '\''s|source=.*|\
+source=("ftp://invisible-island.net/mawk/$pkgname-${pkgver//_/-}.tgz")|'\''
+        fi
+        if pacman -Q "$PACKAGE" 2>/dev/null | \
+                    grep "^$PACKAGE $pkgver-$pkgrel\$"; then
+            echo "Package $PACKAGE already up to date ($pkgver-$pkgrel)."
+        else
+            makepkg -A -s -i '"$*"' --noconfirm
+        fi'
+
+    rm -f "$SUDOERSFILE"
+    undotrap
+
+    return 0
+}
+
+# Detect if multilib repositories are installed
+arch_multilib() {
+    pacman -Sl multilib >/dev/null 2>&1
+}
+
+# install_dist: see install() in prepare.sh for details.
+# We ignore the second part of the parameters, as Arch only pulls required deps.
+# We also replace gcc by its multilib version, if the multilib repository is
+# enabled.
+install_dist() {
+    local pkgs='' aurpkgs='' params='' multilib=''
+    if arch_multilib; then
+        multilib='y'
+    fi
+    while [ "$#" != 0 ]; do
+        if [ "$1" = "--minimal" ]; then
+            true # Ignore: irrelevant for arch
+        elif [ "$1" = "--asdeps" ]; then
+            params="--asdeps"
+        else
+            break
+        fi
+        shift
+    done
+    while [ "$#" != 0 ]; do
+        if [ "$1" = '--' ]; then
+            shift
+            break
+        fi
+        aurstrip=${1#aur:}
+        if [ "$aurstrip" = "$1" ]; then
+            if [ -n "$multilib" ] && \
+	       [ "$1" = "gcc" ]; then
+                pkgs="$pkgs $1-multilib"
+            else
+                pkgs="$pkgs $1"
+            fi
+        else
+            aurpkgs="$aurpkgs $aurstrip"
+        fi
+        shift
+    done
+    if [ -n "$pkgs" ]; then
+        pacman -S --noconfirm --needed $params $pkgs
+    fi
+
+    if [ -n "$aurpkgs" ]; then
+        for aurpkg in $aurpkgs; do
+            aurcompile $aurpkg $params
+        done
+    fi
+}
+
+
+# install_pkg_dist: see install_pkg() in prepare.sh for details.
+install_pkg_dist() {
+    error 1 'ERROR: install_pkg_dist is not supported on Arch'
+}
+
+
+# remove_dist: see remove() in prepare.sh for details.
+remove_dist() {
+    if [ "$#" -gt 0 ]; then
+        pacman -R "$@"
+    fi
+}
+
+
+# list_uninstalled_dist: see list_uninstalled() in prepare.sh for details.
+list_uninstalled_dist() {
+    local suffix="$1" pkg
+    if [ -n "$suffix" ]; then
+        error 1 'list_uninstalled_dist: suffix makes no sense on Arch'
+    fi
+    shift
+    for pkg in "$@"; do
+        if ! pacman -Q "$pkg" 2>/dev/null >/dev/null; then
+            echo -n "$pkg "
+        fi
+    done
+}
+
+# Make sure package architecture is set manually.
+# Using the default Auto here detects the chroot from uname -m, breaking i686
+# chroots on x86-64.
+sed -i 's/^[ \t]*Architecture[ \t].*/Architecture = '"$ARCH"'/' /etc/pacman.conf
```

Comment:
Moved comment to boostrap, and removed the line here. I think this was meant to possibly recover from this issue: https://github.com/drinkcat/chroagh/issues/35. But if this ever becomes necessary, the chroot is probably beyond recovery already (some x86_64 packages were probably installed)